master/internal/trial.go

```diff
@@ -363,58 +371,107 @@ func (t *trial) runningReceive(ctx *actor.Context) error {
 			return err
 		}
 
-	case actor.ChildFailed:
-		ctx.Tell(t.rp, scheduler.TerminateTask{TaskID: t.task.ID, Forcible: true})
+	case actor.ChildFailed, killTrial:
+		t.terminate(ctx, true)
+
+	case allReadyTimeout:
+		if time.Now().After(t.lastContainerConnectedTime.Add(allReadyTimeoutPeriod)) {
+			ctx.Tell(
+				t.logger,
+				model.TrialLog{
+					TrialID: t.id,
+					Message: "waited too long for all containers to be ready; " +
```

Comment:
non-blocking: I would probably tweak it to something like this:
```
some containers are taking a long time to connect to master; when running on kubernetes this may 
happen because only some of the pods have been scheduled; it is possible that some pods will 
never be scheduled without adding compute resources or pausing / killing experiments. 
```

Reply:
Do you mean if you cancel/kill experiments your pods might be scheduled?

Reply:
I mean that if you kill some other experiments this one might get scheduler, but might be too complicated to explain. Maybe we should just suggest killing this experiment and restarting when we know there is enough capacity.